panther_diagnostics/CMakeLists.txt

```
@@ -51,19 +51,32 @@ install(TARGETS system_monitor_node DESTINATION lib/${PROJECT_NAME})
 install(DIRECTORY config launch DESTINATION share/${PROJECT_NAME})
 
 if(BUILD_TESTING)
-  find_package(ament_cmake_gtest REQUIRED)
-
-  ament_add_gtest(${PROJECT_NAME}_test_system_monitor
-                  test/test_system_monitor_node.cpp src/system_monitor_node.cpp)
-  target_include_directories(
-    ${PROJECT_NAME}_test_system_monitor
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-           $<INSTALL_INTERFACE:include> ${CMAKE_INSTALL_PREFIX}/include)
+  find_package(ament_cmake_gmock REQUIRED)
+  find_package(ros_testing REQUIRED)
+
+  # Unit tests
+  ament_add_gmock(
+    ${PROJECT_NAME}_test_system_monitor test/unit/test_system_monitor_node.cpp
+    src/system_monitor_node.cpp)
+  target_include_directories(${PROJECT_NAME}_test_system_monitor
+                             PUBLIC ${CMAKE_INSTALL_PREFIX}/include)
   ament_target_dependencies(${PROJECT_NAME}_test_system_monitor
-                            ${PACKAGE_DEPENDENCIES})
+                            ament_cmake_gtest ${PACKAGE_DEPENDENCIES})
   target_link_libraries(${PROJECT_NAME}_test_system_monitor
                         system_monitor_parameters PkgConfig::CPPUPROFILE)
 
+  ament_add_gmock(${PROJECT_NAME}_test_filesystem test/unit/test_filesystem.cpp)
+  target_include_directories(${PROJECT_NAME}_test_filesystem
+                             PUBLIC ${CMAKE_INSTALL_PREFIX}/include)
+  ament_target_dependencies(${PROJECT_NAME}_test_filesystem ament_cmake_gtest
+                            ${PACKAGE_DEPENDENCIES})
+
+  # Integration tests
+  option(test_integration "Run integration tests" ON)
+  if(test_integration)
```

```suggestion
  option(TEST_INTEGRATION "Run integration tests" ON)
  if(TEST_INTEGRATION)
```
Other variables are capital letters, maybe even `INTEGRATION_TESTING` to make it similar to `BUILD_TESTING`?